lib/std/rand.zig

```diff
@@ -247,18 +247,34 @@ pub const Random = struct {
 
     /// Return a floating point value evenly distributed in the range [0, 1).
     pub fn float(r: Random, comptime T: type) T {
-        // Generate a uniform value between [1, 2) and scale down to [0, 1).
-        // Note: The lowest mantissa bit is always set to 0 so we only use half the available range.
+        // Generate a uniformly random value between for the mantissa.
+        // Then generate an exponentially biased random value for the exponent.
+        // Over the previous method, this has the advantage of being able to represent
+        // much more of the available range.
         switch (T) {
             f32 => {
-                const s = r.int(u32);
-                const repr = (0x7f << 23) | (s >> 9);
-                return @bitCast(f32, repr) - 1.0;
+                // Use the lowest 23 bits for the mantissa, and use the rest to calculate
+                // the exponent. Using only 41 bits for calculating the exponent instead
+                // of the full 128 means that the minimum generatable value is 2^-42, but
+                // it minimizes entropy waste and avoids branching.
+                const rand = r.int(u64);
+                const rand_lz = @clz(u64, rand | 0x7FFFFF);
+                const mantissa = @truncate(u32, rand & 0x7FFFFF);
```

Comment:
the `&` here is redundant with the `@truncate`.

either use (preferred)
```suggestion
                const mantissa = @truncate(u23, rand);
```
or
```suggestion
                const mantissa = @intCast(u32, rand & 0x7FFFFF);
```